documentation/design-docs/dotnet-tools.md

```diff
@@ -396,6 +397,63 @@ CONVERT
       Writing:       ./trace.speedscope.json
       Conversion complete
 
+
+CONVERT
+
+    dotnet-trace stack -p|--process-id <pid>
+                       [-h|--help]
+
+    Prints the stack for every thread in the target process
+
+    -h, --help
+        Show command line help
+
+    Examples:
+      > dotnet-trace stack -p 1234
+      Stack for Thread (2207382):
+          UNMANAGED_CODE_TIME
+          System.Private.CoreLib!System.Threading.ManualResetEventSlim.Wait(int32,value class System.Threading.CancellationToken)
```

Comment:
Devs would probably appreciate more C#-like syntax formatting. Are we formatting this string in the tool or it is being produced by the runtime? I wouldn't want to do anything that takes a lot of dev effort but if the tool formats the string its probably not too hard to pretty some of this stuff up:
- C# keywords for primitive types
- eliminate class/value class terminology
- '.' instead of '+' for inner class
- List\<Foo> instead of List`1[Foo]

Async stacks probably also don't look great but improving them is likely a more substantial task that we wouldn't need to take on right away.

Reply:
> Are we formatting this string in the tool or it is being produced by the runtime?

This is the default output from the TraceEvent library I used for symbol resolution.  I haven't investigated symbolicating enough to know how expensive it would be to get a more C# style representation.

https://github.com/josalem/DotStack/blob/2ddc831df6f7aaa9111f8eb6246bb80ace896610/DotStack.cs#L103-L113